File: wk6/java/throwns.java
// problem link: https://open.kattis.com/problems/throwns

import java.io.*;
import java.util.*;
public class throwns {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        int[] nk = Arrays.stream(r.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        String[] c = r.readLine().split(" ");
        Stack<Integer> s = new Stack<>();
        for(int i = 0; i < c.length; i++) {
            if(c[i].equals("undo")) {
                for(int j = Integer.parseInt(c[i + 1]); j > 0; j--)
                    s.pop();
            i++;
        }
            else
                s.push(Integer.parseInt(c[i]));
        }
        int ret = 0;
        while(!s.isEmpty())
            ret += s.pop();
        while(ret < 0)
            ret += nk[0];
        ret = ret % nk[0];
        System.out.println(ret);
    }
}

File: wk6/py/throwns.py
n, k = map(int, input().split())
commands = input().split()
stack = []
for command in commands:
    if command == "undo":
        stack.append("undo")
        continue

    if not stack or stack[-1] != "undo":
        stack.append(int(command))
    else:
        for i in range(int(command)+1):
            stack.pop()

print(sum(stack)%n)
